### DISCUSSION TITLE
draw_mouse_cursor code

### DISCUSSION THREAD

**Discussion by defdefred** [VendelinSlezak/BleskOS#28] [Ideas]
About:
```
void draw_mouse_cursor(dword_t x, dword_t y) {
 dword_t *screen;
 dword_t first_line_pixel_pointer = (screen_mem + (y*screen_bytes_per_line) + (x<<2));
 dword_t old_x = x;
 dword_t *mouse_cursor_img_ptr = (dword_t *) mouse_cursor_img;
 
 if(x>graphic_screen_x || y>graphic_screen_y) {
  return;
 }
```
I wonder if moving the test for out-of-screen mouse before the whole variable declaration could help saving some cpu resources?

And I have an idea to optimize the cursor drawing...
How about reversing the cursor when reaching the right border minus cursor-width to allow always full  drawing with optimized code? Is it stupid?

**Comment by VendelinSlezak** [comment]
> I wonder if moving the test for out-of-screen mouse before the whole variable declaration could help saving some cpu resources?

Yeah, sure, I will fix it in next update.

> How about reversing the cursor when reaching the right border minus cursor-width to allow always full drawing with optimized code? Is it stupid?

Well, cursor is so small that current code is so fast that such change would no really improve performace, but I think it would look strange that when you would go with cursor to the border suddenly it would turn horizontally or even vertically.

**Comment by VendelinSlezak** [comment]
I implemented this in Update 13.
